OpenRA.Game/Widgets/Widget.cs

```diff
@@ -179,15 +191,15 @@ public abstract class Widget
 		public string Height = "0";
 		public string[] Logic = { };
 		public ChromeLogic[] LogicObjects { get; private set; }
-		public bool Visible = true;
+		public bool Visible { set { VisibilityFunction = () => value; } }
 		public bool IgnoreMouseOver;
 		public bool IgnoreChildMouseOver;
 
-		// Calculated internally
-		public Rectangle Bounds;
-		public Widget Parent = null;
-		public Func<bool> IsVisible;
-		public Widget() { IsVisible = () => Visible; }
+		private Widget parent = null;
+		public Widget Parent { get { return parent; } }
+		private Func<bool> visibilityFunction;
+		public Func<bool> VisibilityFunction { get { return visibilityFunction; } set { visibilityFunction = value; IsVisible(); } }
```

Comment:
Having this and `IsVisible` as distinct things feels quite weird IMO, but I don't have any immediate alternatives to suggest.

Reply:
The main problem here is to allow the old ui to work unaltered. Over time, the old ui will be refactored to rely on yoga features only and we can get rid of those workarounds. In the end this can be cleaned up when we no longer need to provide maximum compatibility with the old setup. The problem is actualy, that Nodes must be visible in order to calculate their layout bounds, which is totaly fine on a pure yoga layout, but as long as we manualy set and directly use bounds for further bounds calculations, i had to hack in several visibility and layouting calls to make sure the old ui properly works. the perfect example is a dropdown, which template widget is not visible, resulting in all clones to not have any bounds. As as conclusion, the end-goal would be to get rid of all the manual calculations, and to refactor the visibility code towards updating the display prop in the layout method.

Reply:
My expectation/assumption was that clones would copy the string-based definitions (`X`, `Width`, etc) and then set their layout based on these values rather than the computed layout of the template. Is there something stopping this from working?

Reply:
Is it possible to keep the current `IsVisible` stuff as it is, but then introduce a new `ComputeLayout` pass to `Widget` that evaluates `IsVisible` to update the yoga visibility and recompute layout before rendering?